internal/controllers/synthesis/lifecycle.go

```diff
@@ -197,6 +150,41 @@ func (c *podLifecycleController) Reconcile(ctx context.Context, req ctrl.Request
 	return ctrl.Result{}, nil
 }
 
+// safeToCreatePod returns true when it is safe to create a new synthesizer pod.
+func (c *podLifecycleController) safeToCreatePod(ctx context.Context, pods *corev1.PodList, comp *apiv1.Composition) (bool, error) {
+	logger := logr.FromContextOrDiscard(ctx)
+	sort.Slice(pods.Items, func(i, j int) bool { return pods.Items[i].CreationTimestamp.Before(&pods.Items[j].CreationTimestamp) })
+
+	var active bool
+	var deleting bool
+	for _, pod := range pods.Items {
+		if pod.DeletionTimestamp != nil {
+			if !deleting {
+				deleting = true
+				continue
+			}
+			return false, nil // don't create new pod if two are deleting
+		}
+
+		if podIsCurrent(comp, &pod) {
+			if !active {
+				active = true
+				continue
+			}
+
+			// Clean up any dupes
+			err := c.client.Delete(ctx, &pod)
```

Comment:
There is one deletion case that can't logically be addressed by the podGC controller

Reply:
It feels weird to have this here, why can't it be done in the gc controller?

Reply:
Hmm let me see what the tradeoffs would be

Reply:
Remove it! Turns out we don't really need it anyway. I reworked a few things to avoid even needing a pod index.